diff --git a/main/lib/handlers/handleWindowLifecycleEvents.js b/main/lib/handlers/handleWindowLifecycleEvents.js
--- a/main/lib/handlers/handleWindowLifecycleEvents.js
+++ b/main/lib/handlers/handleWindowLifecycleEvents.js
@@ -56,4 +56,7 @@ const handleWindowLifecycleEvents = (window) => {
     window.on('close', (event) => {
+        if (isTrayEnabled && !state_js_1.state.player.isPlaying) {
+            electron_1.app.quit();
+        }
         if (!isTrayEnabled) {
             electron_1.app.quit();
         }
